### PR TITLE
pillow 10.4.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-5184](https://anaconda.atlassian.net/browse/PKG-5184) 
- [Upstream repository](https://github.com/python-pillow/Pillow/blob/10.4.0)
- [Diff between the latest and previous upstream releases](https://github.com/python-pillow/Pillow/compare/10.3.0...10.4.0)
- Changelog: 
  - https://github.com/python-pillow/Pillow/blob/10.4.0/CHANGES.rst
  - https://pillow.readthedocs.io/en/stable/releasenotes/10.4.0.html
  
- Requirements:
  - https://github.com/python-pillow/Pillow/blob/10.4.0/pyproject.toml
  - https://github.com/python-pillow/Pillow/blob/10.4.0/setup.py
  - https://pillow.readthedocs.io/en/stable/installation/building-from-source.html#building-from-source

### Explanation of changes:

- Add `PIL._imaging` to `test/imports` to prevent an error `DLL load failed while importing _imaging`

[PKG-5184]: https://anaconda.atlassian.net/browse/PKG-5184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -48,6 +48,8 @@ test:
     - PIL
     - PIL.Image
     - PIL.ImageCms  # [not win]
+    # Testing _imaging to prevent an error 'DLL load failed while importing _imaging'
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:51]:
```suggestion
    # Testing _imaging to verify that it doesn't fail like reported in https://anaconda.atlassian.net/browse/PKG-5184
```

**Comment by lorepirri** [comment]
> `Pillow >= 2.1.0` no longer supports `import _imaging`. Please use `from PIL.Image import core as _imaging` instead.
> versionremoved:: 2.1.0

https://github.com/python-pillow/Pillow/blob/6a9acfa5ca2b3ba462b086e4af776993bbd94a72/docs/deprecations.rst#import-_imaging

File: recipe/meta.yaml [hunk 2]
```diff
@@ -48,6 +48,8 @@ test:
     - PIL
     - PIL.Image
     - PIL.ImageCms  # [not win]
+    # Testing _imaging to verify that it doesn't fail like reported in https://anaconda.atlassian.net/browse/PKG-5184
+    - PIL._imaging
```
Comment (lorepirri) [inline recipe/meta.yaml:52]: You could add a test command `python -c "from PIL.Image import core"`

Comment (lorepirri) [reply recipe/meta.yaml:52]: The problem was in `openjpeg 2.4.0=*_1` see https://anaconda.slack.com/archives/C02D68R4D0D/p1720646319721279?thread_ts=1719980913.368079&cid=C02D68R4D0D

Comment (skupr-anaconda) [reply recipe/meta.yaml:52]: I added `python -c "from PIL.Image import core"`

**Review by lorepirri — APPROVED** [review]

**Comment by danpetry** [comment]
AFAIU the problem was in openjpeg, and the added tests/comments would be misleading?

**Comment by lorepirri** [comment]
> AFAIU the problem was in openjpeg, and the added tests/comments would be misleading?

the link to the ticket will be helpful: the next time we are pointed at `_imaging` we could figure that it might be another thing, tested, and there is a ticket reference to follow. This problem is reported in other places.

**Review by danpetry — APPROVED** [review]
ok, I don't object strongly. Just pointing out that if we test `PIL._imaging`, then following the same principle we should test _everything_, because any other library could also have a broken .dll (or whatever it was). if you see what I mean.

**Comment by cbouss** [comment]
Besides this test wouldn't detect it because of the our image having vc runtime debug libraries.

**Comment by cbouss** [comment]
> > Besides this test wouldn't detect it because of the our image having vc runtime debug libraries.
> 
> Our instance has the debug symbols, but they won't be accessible during tests because the the compiler activation package is not part of the packages included in the tests.



> > Besides this test wouldn't detect it because of the our image having vc runtime debug libraries.
> 
> Our instance has the debug symbols, but they won't be accessible during tests because the the compiler activation package is not part of the packages included in the tests.

The vc runtime is in system32, so it is found, activation scripts or not.

**Comment by JeanChristopheMorinPerso** [comment]
But aren't the debug libs only in the msvc install?

**Comment by cbouss** [comment]
> But aren't the debug libs only in the msvc install?

The debug dlls are in system32. So that's what the previous build of openjpeg was getting.

**Pull request merged**